src/sonic-eventd/rsyslog_plugin/rsyslog_plugin.cpp

```diff
@@ -106,9 +106,8 @@ bool RsyslogPlugin::createRegexList() {
 void RsyslogPlugin::run() {
     lua_State* luaState = luaL_newstate();
     luaL_openlibs(luaState);
-    while(true) {
-        string line;
-        getline(cin, line);
+    string line;
+    while(getline(cin, line)) {
```

Comment:
I'd recommend adding a signal handler in rsyslog_plugin. something in the lines of 

```
trap SIGTERM signal_handler

void signal_handler(){
        stop_processing = true;
}

while(!stop_processing && !getline(cin,buf)){
       eventd logic.....
}
```

rsyslog dameon first sends a SIGTERM which would cause the program to terminate immediately. Meanwhile the plugin might be processing something it read just before from cin. So, to let it finish the pending task i'd suggest this approach

Reply:
Added signal handler.